handlers/containers.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+)
+
+type Function struct {
+	name      string
+	namespace string
+	image     string
+	pid       uint32
+	replicas  int
+	IP        string
+}
+
+const (
+	// FunctionNamespace is the containerd namespace functions are created
+	FunctionNamespace = "openfaas-fn"
+)
+
+// ListFunctions returns a map of all functions with running tasks on namespace
+func ListFunctions(client *containerd.Client) (map[string]Function, error) {
+	ctx := namespaces.WithNamespace(context.Background(), FunctionNamespace)
+	functions := make(map[string]Function)
+
+	containers, _ := client.Containers(ctx)
+	for _, k := range containers {
+		name := k.ID()
+		functions[name], _ = GetFunction(client, name)
+	}
+	return functions, nil
+}
+
+// GetFunction returns a function that matches name
+func GetFunction(client *containerd.Client, name string) (Function, error) {
+	ctx := namespaces.WithNamespace(context.Background(), FunctionNamespace)
+	c, err := client.LoadContainer(ctx, name)
+	if err == nil {
+
+		image, _ := c.Image(ctx)
+		f := Function{
+			name:      c.ID(),
+			namespace: FunctionNamespace,
+			image:     image.Name(),
+		}
+		replicas := 0
+		task, err := c.Task(ctx, nil)
```

Comment:
If the task is in a frozen state, the replicas should also be 0, this could be fixed in a follow-up PR